### PR TITLE
refactor(hydroflow_lang): update topo-sort to detect cycles

### PR DESCRIPTION
**Pull request by MingweiSamuel**

### CODE REVIEW

File: hydroflow_lang/src/graph/graph_algorithms.rs [hunk 1]
```diff
@@ -187,6 +225,47 @@ mod test {
         }
     }
 
+    #[test]
+    pub fn test_toposort_cycle() {
+        // https://commons.wikimedia.org/wiki/File:Directed_graph,_cyclic.svg
```
Comment (rohitkulshreshtha) [inline hydroflow_lang/src/graph/graph_algorithms.rs:230]: Nit: this isn't a permalink.

Comment (rohitkulshreshtha) [reply hydroflow_lang/src/graph/graph_algorithms.rs:230]: If it helps:

```
         ┌────►C──────┐       
         │            │       
         │            ▼       
A───────►B            E ─────►F
         ▲            │       
         │            │       
         └─────D◄─────┘       
```

**Review by rohitkulshreshtha — APPROVED** [review]
LGTM!

File: hydroflow_lang/src/graph/graph_algorithms.rs (hunk 1 above)
Comment (MingweiSamuel) [reply hydroflow_lang/src/graph/graph_algorithms.rs:230]: Nice haha

**Pull request merged**